src/components/map/MapboxGlMap.jsx

```diff
@@ -97,6 +97,16 @@ export default class MapboxGlMap extends React.Component {
     }
   }
 
+  shouldComponentUpdate(nextProps, nextState) {
+    let should = false;
+    try {
+      should = JSON.stringify(this.props) !== JSON.stringify(nextProps) || JSON.stringify(this.state) !== JSON.stringify(nextState);
```

Comment:
This can possibly block updates via `setState` - `this.state` contains non-JSON-serializable data (at least `map`), which means `should` is only true if the props change. If the props are the same, the second part of the ORed expression will be executed and throw an error. Not a problem currently, as nothing is relying on updates via `setState`, but may be come a problem later.

I don't fully understand how to fix it though - maybe don't keep `map`/`inspect` in state, and as a property instead?